Feature: add stale flag in request return object

MerryCello opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review the requirements below.

Bug fixes and new features should include tests.

By making a contribution to this project, I certify that:

* (a) The contribution was created in whole or in part by me and I
  have the right to submit it under the open source license
  indicated in the file; or

* (b) The contribution is based upon previous work that, to the best
  of my knowledge, is covered under an appropriate open source
  license and I have the right under that license to submit that
  work with modifications, whether created in whole or in part
  by me, under the same open source license (unless I am
  permitted to submit under a different license), as indicated
  in the file; or

* (c) The contribution was provided directly to me by some other
  person who certified (a), (b) or (c) and I have not modified
  it.

* (d) I understand and agree that this project and the contribution
  are public and that a record of the contribution (including all
  personal information I submit with it, including my sign-off) is
  maintained indefinitely and may be redistributed consistent with
  this project or the open source license(s) involved.
-->
This PR is an attempt to add the request response "stale" flag that is present in the axios-cache-adapter. This flag says whether or not the response is from staled cache. This would be returned along with the "cached" flag. Please let me know if there's anything that I'm missing.

arthurfiorette commented on src/cache/axios.ts:
```diff
@@ -44,6 +44,13 @@ export interface CacheAxiosResponse<R = any, D = any> extends AxiosResponse<R, D
+   * 
+   * @see https://axios-cache-interceptor.js.org/config/response-object#stale
+   */
+  stale: boolean;
```
this type might be undefined in some areas, probably a `stale?: boolean` is a better type

arthurfiorette commented on src/interceptors/response.ts:
```diff
@@ -348,6 +348,7 @@ export function defaultResponseInterceptor(axios: AxiosCacheInstance): ResponseI
 
         return {
           cached: true,
+          stale: true,
```
you can't you use .cached here? since this boolean will always be true when cached is true too?

MerryCello replied:
I see your point. In other cases stale may be undefined/false when cached is true.

MerryCello replied:
Are there other cases where cache is true and the state is stale?

arthurfiorette replied:
What I meant was:

why `response.cached` cannot be used in your case? Since both these values will always have the same value?

MerryCello replied:
They will be the same when staleIfError is set, but is other cases `response.stale` may be falsy when `response.cached` is true. For example: when a request is made, and the cache is still valid (i.e. now() < ttl), then the cache is returned with `{...repsonse, cached: true, stale: false}`. Does that make sense?

MerryCello replied:
@arthurfiorette, I added more tests in cases where `cached: true` and `stale: false` to show where `stale` won't always have the same value as `cached`. This might make it a little more clear. Please review and lmk what you think.

MerryCello replied:
Hi @arthurfiorette, just following up on this since it's been a while. Let me know if you have further questions.

arthurfiorette commented:
please also add tests

arthurfiorette commented:
Ci is failing

MerryCello commented:
@arthurfiorette, I added some tests and fixed the failing ones. I also added some comments as replies above that may have been missed. If you could review those, that would be great.

arthurfiorette commented:
thanks for your work!
